backend/assignments/1/src/main/java/jav/assignment/Transactions/BuyTransaction.java

```diff
@@ -0,0 +1,62 @@
+package jav.assignment.Transactions;
+
+import jav.assignment.Coins.Coin;
+import jav.assignment.BlockHash.BlockHash;
+import jav.assignment.Entities.Buy;
+import java.util.concurrent.CountDownLatch;
+
+import static jav.assignment.globalObjects.GlobalData.tradersData;
+public class BuyTransaction extends Thread {
+    private Buy buy;
+    private String transactionID;
+    private CountDownLatch latch;  // Add this field
+
+    public BuyTransaction(Buy buy, CountDownLatch latch) {
+        this.buy = buy;
+        this.latch = latch;
+    }
+
+    @Override
+    public void run() {
+        synchronized (this.buy.getCoin()) {
+            try {
+                while (this.buy.getCoin().getStatus().equals("not-available") || this.buy.getCoin().getCirculatingSupply() < this.buy.getQuantity()) {
```

Comment:
Move status strings : available, not-available
AND
Fixed opertions : BUY, SELL etc 
To constants